### ISSUE TITLE
Cabal test is not using the same flags as cabal install?

### ISSUE THREAD

**Issue by rrnewton** [iu-parfunc/lvars#95]
In [logs like this one](http://tester-lin.soic.indiana.edu:8080/view/ParfuncTests/job/LVish-implementation-2.0/263/CABAL=cabal-1.20,CABAL_FLAGS1=-f-debug,CABAL_FLAGS2=-f-getonce,CABAL_FLAGS3=-f-threaded,CABAL_FLAGS4=-fchaselev,CABAL_FLAGS5=-f-useopt,JENKINS_GHC=7.8.3,PROF=0,label=linux-soic/console), we see the following:

```
v3e: [Failed]
ERROR: Final continuation of Par computation was duplicated, in spite of GET_ONCE!
```

In spite of the fact that getonce should be turned OFF on that run, and we clearly see that in the options passed to `cabal install`:

``` bash
+ cabal-1.20 install --force-reinstalls --disable-library-profiling --disable-executable-profiling -f-debug -f-getonce -f-threaded -fchaselev -f-useopt --with-ghc=ghc-7.8.3 ./lvish ./par-classes ./par-collections ./par-transformers ./monad-par/monad-par/ --enable-tests -j
Resolving dependencies...
```

Basically, it's clear what the problem is.  Cabal test is rebuilding the libraries even though it shouldn't:

```
+ cabal-1.20 test --show-details=streaming '--test-options=-j1 --jxml=test-results.xml --jxml-nested'
Package has never been configured. Configuring with default flags. If this
fails, please run configure manually.
Resolving dependencies...
Configuring lvish-2.0...
Building lvish-2.0...
Preprocessing library lvish-2.0...
```

And therefore it just drops the flags.  WHY is cabal test rebuilding when we just build everything?  I have no idea.  But it's clear that `cabal test` is not satisfactory.
